specification/redisenterprise/resource-manager/Microsoft.Cache/preview/2021-02-01-preview/redisenterprise.json

```diff
@@ -0,0 +1,1762 @@
+          "$ref": "#/definitions/DatabaseProperties",
+          "description": "Properties of the database."
+        }
+      }
+    },
+    "DatabaseProperties": {
+      "title": "RedisEnterprise database properties",
+      "description": "Properties of RedisEnterprise databases, as opposed to general resource properties like location, tags",
+      "properties": {
+        "clientProtocol": {
+          "type": "string",
+          "description": "Specifies whether redis clients can connect using TLS-encrypted or plaintext redis protocols. Default is TLS-encrypted.",
+          "enum": [
+            "Encrypted",
+            "Plaintext"
+          ],
+          "x-ms-enum": {
+            "name": "Protocol",
+            "modelAsString": true
+          }
+        },
+        "port": {
+          "type": "integer",
+          "format": "int32",
+          "x-ms-mutability": [
+            "read",
+            "create"
+          ],
+          "description": "TCP port of the database endpoint. Specified at create time. Defaults to an available port."
+        },
+        "provisioningState": {
+          "$ref": "#/definitions/ProvisioningState",
+          "readOnly": true,
+          "description": "Current provisioning status of the database"
+        },
+        "resourceState": {
+          "$ref": "#/definitions/ResourceState",
+          "readOnly": true,
+          "description": "Current resource status of the database"
+        },
+        "clusteringPolicy": {
+          "type": "string",
+          "description": "Clustering policy - default is OSSCluster. Specified at create time.",
+          "enum": [
+            "EnterpriseCluster",
+            "OSSCluster"
+          ],
+          "x-ms-enum": {
+            "name": "ClusteringPolicy",
+            "modelAsString": true
+          },
+          "x-ms-mutability": [
+            "read",
+            "create"
+          ]
+        },
+        "evictionPolicy": {
+          "type": "string",
+          "description": "Redis eviction policy - default is VolatileLRU",
+          "enum": [
+            "AllKeysLFU",
+            "AllKeysLRU",
+            "AllKeysRandom",
+            "VolatileLRU",
+            "VolatileLFU",
+            "VolatileTTL",
+            "VolatileRandom",
+            "NoEviction"
+          ],
+          "x-ms-enum": {
+            "name": "EvictionPolicy",
+            "modelAsString": true
+          }
+        },
+        "persistence": {
+          "$ref": "#/definitions/Persistence",
+          "description": "Persistence settings"
+        },
+        "modules": {
+          "type": "array",
+          "description": "Optional set of redis modules to enable in this database - modules can only be added at creation time.",
+          "items": {
+            "$ref": "#/definitions/Module"
+          },
+          "x-ms-mutability": [
+            "read",
+            "create"
+          ]
+        },
+        "georeplication": {
```

Comment:
camelCase geoReplication to be consistent with other properties?

Reply:
@TimLovellSmith what do you think about this? I know that you had previously given feedback to case it as "georeplication"

Reply:
To add more info on why I thought it might need to be geoReplication:

Documentation shows that this word in English is typically spelled: geo-replication (https://docs.microsoft.com/en-us/azure/azure-sql/database/active-geo-replication-configure-portal)

This type of spelling means this is a hyphenated compound word.

Microsoft guidelines on how to handle compound words: https://docs.microsoft.com/en-us/dotnet/standard/design-guidelines/capitalization-conventions#capitalizing-compound-words-and-common-terms

Therefore, if hyphenated compound words are a subset of compound words, I think "georeplication" is actually the correct style

Reply:
Microsoft guidelines on hyphenated compound words: https://docs.microsoft.com/en-us/style-guide/punctuation/dashes-hyphens/hyphens#capitalization-in-hyphenated-compound-words

Reply:
+1 on using lowerCamelCase for 'geoReplication'

Reply:
I stand corrected!